src/scripts/main.js

```diff
@@ -1,3 +1,29 @@
 'use strict';
 
-// write code here
+const listEl = document.querySelector('ul');
+const listItems = [...listEl.children];
+
+const convertToNumber = str => {
+  if (typeof str === 'string') {
+    const cleanedStr = str.replace(/[^\d.-]/g, '');
+    return Number(cleanedStr);
+  }
+  return NaN;
+};
+
+const sortList = (list, sortParam) => {
+  return [...list].sort((a, b) =>
+    convertToNumber(b.dataset[sortParam]) - convertToNumber(a.dataset[sortParam])
+  );
+};
```

Comment:
This function has a side effect of modifying the original list. It's better to avoid side effects if possible. Consider making a copy of the list before sorting it.